0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md

```diff
@@ -1,47 +1,104 @@
+8. The sender's SPSP client uses its ILP module to initiate the ILP transfer.
+9. The receiver's ILP module registers the incoming transfer held pending the fulfillment of the Crypto Condition. It validates that the transfer matches the packet and regenerates the condition fulfillment using the [Interactive Transport Protocol (ITP)](../0011-interactive-transport-protocol). The ILP module submits the fulfillment to execute the transfer and claim the funds.
+10. The sender's SPSP client receives a notification from its ILP module that the transfer has been executed, including the condition fulfillment from the recipient, and notifies the sender that the payment is completed.
 
-Any SPSP recipient will create a receiving endpoint called the *receiver*. The sender queries this endpoint to get information about the type of payment that can be made to this receiver:
+#### Fixed Source Amount
+
+SPSP may be used by the sender to send a payment of a fixed amount of the sender's chosen currency or asset type. This is illustrated with the following example:
+
+We assume that the sender knows the receiver endpoint (see [Appendix A: (Optional) Webfinger Discovery](#appendix-a-optional-webfinger-discovery)).
+
+1. The sender's SPSP client queries the receiver endpoint.
+2. The receiver endpoint responds with the receiver info, including the recipient's ILP address.
+3. The sender choses an amount of their currency or assets to send.
+4. The sender's SPSP client uses its ILP module to quote how much of the recipient's currency or asset type will be delivered to the recipient's ILP address for the given source amount.
+5. The sender accepts the quote.
+6. The sender's SPSP client submits the payment information, including the destination amount, to the receiver endpoint.
+7. The receiver endpoint responds with an ILP Packet and Crypto Condition corresponding to the destination amount.
+8. The sender's SPSP client uses its ILP module to create a transfer **using the chosen source amount, NOT by quoting the ILP packet**, and attach the ILP packet to the transfer.
+9. The receiver's ILP module registers the incoming transfer held pending the fulfillment of the Crypto Condition. It validates that the transfer matches the packet and regenerates the condition fulfillment using the [Interactive Transport Protocol (ITP)](../0011-interactive-transport-protocol). The ILP module submits the fulfillment to execute the transfer and claim the funds.
+10. The sender's SPSP client receives a notification from its ILP module that the transfer has been executed, including the condition fulfillment from the recipient, and notifies the sender that the payment is completed.
+
+#### Invoice
+
+1. The sender's SPSP client queries the receiver endpoint.
+2. The receiver endpoint responds with the receiver info, including the invoice status, amount, and currency code.
+3. The sender's SPSP client submits the sender's info, including the sender's ILP address to the receiver endpoint.
```

Comment:
@justmoon what do you think the client should submit to the receiver endpoint for an invoice? An empty `POST` or some information about themselves? I put the sender's ILP address in case the receiver might have the sender send to a different ILP address depending on where they're sending from.